### ISSUE TITLE
v8 v7.8 rolling-origin conformal positive-EV action policy

### ISSUE THREAD

**Issue by phead198708** [phead198708/BiGan#242]
## Goal

Create a new v8 candidate that maximizes execution-realizable fixed-size after-cost PnL by requiring a strictly prior, uncertainty-adjusted positive expected-return lower bound before retaining or switching an SBC action.

Candidate:

`rolling_origin_conformal_positive_ev_v7_8`

This issue starts a new development lineage. The terminal #241 future-holdout outcomes are excluded from fitting, calibration, threshold selection, feature selection, model selection, and gate design.

## Motivation and scope

v7.7 established the rolling-origin, same-decision, no-side-quota framework. v7.8 retains that framework but replaces the fixed point-estimate edge rule with a preregistered one-sided conformal lower-bound rule.

This is a historical-development candidate first. No new collection is allowed unless the frozen historical replay is positive after costs and noninferior to v6.7.

## Frozen historical inputs

Use only lineage available before #241 target access:

- canonical 134-market runtime-aligned seed corpus
- consumed 120-market historical development stream used by #240
- frozen v6.2 source candidate
- frozen v6.7 baseline/profile
- frozen runtime cost, size, execution guard, and position-management contract
- existing canonical decision-time features

Explicitly forbidden inputs:

- #241 settled index, market outcomes, PnL rows, side/action diagnostics, or gate result
- #239 outer-OOF metrics/rows
- #238 side/action/PnL attribution
- any current validation/holdout PnL used for feature, coefficient, quantile, support, or threshold tuning

## Model contract

Keep the v7.7 joint SBC action-value model unchanged:

- one weighted XGBoost regressor
- actions: `BUY_UP_SELL_BEFORE_CLOSE`, `BUY_DOWN_SELL_BEFORE_CLOSE`
- target: runtime-aligned after-cost net return per contract
- exponential decay half-life: `60` markets
- minimum model-training support: `40` markets
- existing v7.7 XGBoost parameters unchanged
- no hyperparameter/profile/window search
- no side-specific model or quota
- no HTS activation
- maximum one accepted bet per market

## Strictly prior conformal calibration

Use one-sided overprediction residuals:

```text
action_overprediction_residual = predicted_return - realized_after_cost_return
paired_delta_overprediction_residual = predicted_opposite_minus_baseline - realized_opposite_minus_baseline
```

Calibration protocol:

1. Order markets by decision timestamp, then market id.
2. Build initial residual history from rolling-origin predictions on the seed corpus: first 40 seed markets initialize training; each later seed market is predicted before its target enters the next fit.
3. During the 120-market development stream, current-market targets are unavailable until its decision is frozen.
4. Add current residuals only after official target availability and market close, for use by later markets.
5. Pool residuals across UP/DOWN; no side weighting or quota.
6. Require at least `60` prior calibration markets before a conformal action is available.
7. Use the deterministic finite-sample one-sided `0.90` quantile: sorted residual index `ceil((n + 1) * 0.90)`, capped at `n`.
8. No quantile/profile search and no result-selected rerun.

Per decision:

```text
action_lcb = predicted_return - q90(action_overprediction_residual)
switch_delta_lcb = predicted_opposite_minus_baseline - q90(paired_delta_overprediction_residual)
```

## Frozen decision rule

Allowed decisions:

- `KEEP_V6_7`
- `SWITCH_SAME_DECISION_SBC`
- `VETO_TO_NO_TRADE`

Rules:

1. A v6.7 `NO_TRADE` market remains `NO_TRADE`; v7.8 cannot activate it.
2. Switch only when opposite `action_lcb > 0` and `switch_delta_lcb > 0`.
3. Otherwise retain v6.7 only when baseline `action_lcb > 0`.
4. Otherwise veto to `NO_TRADE`.
5. Baseline and opposite actions must be from the same market and exact decision timestamp.
6. The unchanged full execution guard is applied after selection.
7. No side quota or side-level hard PnL gate.

## Historical prequential hard gate

Evaluate once on the same exact 120 historical development markets, with each prediction frozen before current-market target access.

Required:

- exact 120 market reconciliation
- candidate guard-accepted unique markets `>= 40`
- candidate fixed-size total after-cost PnL `> 0`
- candidate minus v6.7 total after-cost PnL `>= 0`
- candidate minus v6.7 largest-winner-removed PnL `>= 0`
- at least 3 policy differences for target-free actionability
- complete same-decision five-action/source lineage
- all calibration residuals strictly prior
- feature `max_input_ts <= decision_ts`
- coefficients and residual quantiles finite
- no #241 input access

Equality passes comparative noninferiority. Side/action/family metrics are diagnostic only.

If any historical hard gate fails, stop before new collection.

## Target-free canary

Only after the historical hard gate passes:

- freeze model, conformal state, profile, hashes, and implementation commit
- collect the earliest exact 12 quality-valid strictly-later/disjoint markets within 18 attempts
- minimum market start strictly greater than `1784806800000`, the close of the latest market in the complete #241 collection batch 11
- no outcome/resolution/label/PnL access
- score only after the bounded batch seals
- unchanged execution guard/cost/size
- require at least 1 guard-accepted policy difference and at least 4 guard-accepted markets
- no side quota

If target-free actionability fails, stop before a PnL holdout.

## Future unseen PnL holdout

Only after canary pass, create a separately frozen exact-120 strictly-later outcome-blind holdout within scan cap 180.

Single-use gate:

- candidate guard-accepted unique markets `>= 40`
- exact 120 official read-only settlements on quarantine copies
- candidate total fixed-size after-cost PnL `> 0`
- candidate minus v6.7 total after-cost PnL `>= 0`
- candidate minus v6.7 largest-winner-removed PnL `>= 0`
- no result-dependent rerun, extension, or refit

Passing provides promotion-discussion evidence only; it does not automatically unlock paper/live.

## Required artifacts

- frozen v7.8 profile and hash
- rolling-origin conformal model/state artifact
- historical prequential predictions and residual lineage
- leakage/causality audit
- historical same-market v7.8-v6.7 comparison report/manifest
- target-free canary plan/report/manifest if authorized
- future holdout plan/report/manifest only if canary passes

## Safety

Always keep:

- `paper_only=true`
- `capital_at_risk=false`
- `polymarket_write_enabled=false`
- `wallet_signing_enabled=false`
- `v8_execution_handoff_allowed=false`
- `source_model_candidate_eligible=false`
- `freeze_ready=false`
- `promotion_evidence_eligible=false`
- `paper_candidate_allowed=false`
- `#134_resume_allowed=false`
- `#146_start_allowed=false`

No live trading, wallet signing, Polymarket writes, score mutation, production gate relaxation, or use of current OOF/validation/holdout PnL for tuning.

**Comment by phead198708** [comment]
Implementation freeze before historical target access:

- branch: `codex/v8-issue-242-conformal-positive-ev`
- implementation commit: `aad23ee` (`aad23ee` is the pushed branch HEAD; the full SHA is bound in the generated artifact)
- v7.8 profile SHA-256: `51519cacf9e190701988cdde21531bf815850a23bda187d667cf95838161b205`
- focused v7.8 tests: `5 passed`
- v7.7 regression tests: `11 passed`
- Ruff: passed
- `git diff --check`: passed
- `git diff --cached --check`: passed before commit

The next action is the preregistered single historical exact-120 prequential hard gate. It will use only pre-#241 lineage. The #241 settled rows, PnL, diagnostics, and gate result are excluded from fitting, calibration, threshold selection, and gate design. No result-selected rerun is allowed. No collection starts unless this gate passes.

**Comment by phead198708** [comment]
## Terminal historical hard-gate result

The preregistered exact-120 historical prequential gate was executed once from pushed commit `aad23eef8b4ba72999f7499e66071ec82366934c`. No #241 outcome/PnL/gate artifact was accepted as input, and no result-selected rerun was performed.

### Result

- run id: `v7-8-rolling-origin-conformal-positive-ev-historical-20260723T122000Z`
- exact evaluation markets: `120`
- candidate guard-accepted unique markets: `0` (required `>=40`)
- policy distribution: `VETO_TO_NO_TRADE=120`
- candidate fixed-size after-cost PnL: `0.0`
- v6.7 fixed-size after-cost PnL: `-0.09499999999999985`
- candidate minus v6.7 total PnL: `+0.09499999999999985`
- candidate minus v6.7 largest-winner-removed PnL: `+0.21394999999999986`
- historical policy difference markets: `120`
- action-overprediction q90: `0.397535061776638`
- paired-delta-overprediction q90: `0.6151731510460376`
- fit/leakage audit: passed
- timestamp causality: passed
- strictly-prior target/residual chronology: passed
- baseline identity reconciliation: passed

The q90 lower bounds were below zero for every market:

- baseline action LCB max/median/min: `-0.294161 / -0.395227 / -0.491021`
- opposite action LCB max/median/min: `-0.339245 / -0.413529 / -0.605194`

This is a clean fail-closed result: v7.8 is comparatively noninferior only because it places no bets. It fails both executable support and positive candidate PnL, so no target-free canary or new collection is authorized.

Blocking reason codes:

- `historical_candidate_guard_accepted_support_insufficient`
- `historical_candidate_total_after_cost_pnl_not_positive`

### Artifact hashes

- profile: `51519cacf9e190701988cdde21531bf815850a23bda187d667cf95838161b205`
- manifest: `70a8f721fba85ae713e390f5619efbf959c554a7370bce10c96930bf82566669`
- model: `ebbb476148e2339bb0c58a70329c7c5d22391cca60c6a5787a5a707469eac937`
- hard-gate report: `99d62b41e90fa559a99ca0246048c3a56e61cfac9b3ed1c1ef107c99cf70d442`
- leakage audit: `860bae99e16f646d1259b9f3cb7a0327ab207dc5d140313b13dec74399931d47`
- prequential policy rows: `eaf2f7861dc0d6004fe71a704f11f7e7bce474e7b01071e28cd8a170ea8f2819`
- guard replay rows: `80dea7bbf416821c46444ee51f2442a88a23a9d71423c96f1144cbea6285a0fe`
- strictly-prior residual lineage: `e39aa2253b35748ce3ac1ca3eac372c2335a721bac15c5fcd1c4911233bbd9c9`

### Validation

- focused + v7.7 regression suite: `45 passed`
- Ruff (`src/bigan/v8/polymarket examples/v8 tests/v8`): passed
- `git diff --check`: passed
- `git diff --cached --check`: passed

Safety remains fail-closed: paper/live/write/wallet/capital/handoff/source/freeze/promotion/#134/#146 are all false. Historical replay is development evidence only, not promotion evidence.

**Comment by phead198708** [comment]
Closing after the preregistered historical gate completed terminally. The candidate was correctly rejected before any new collection; no safety or promotion gate was unlocked.
